### PR TITLE
feat: improve segment edit

### PR DESCRIPTION
**Pull request by chubetho**
User now can
-  see detail of each segment in edit modal
- edit metadata
- create new metadata based on current selected metadata

### CODE REVIEW

File: src/frontend/components/ModalEditSegment.vue [hunk 1]
```diff
@@ -26,11 +35,92 @@ const emits = defineEmits<{
...
+const currentProcess = ref(0)
+const currentTime = ref(0)
+const duration = ref(1)
+
 const isMouseOverRow = ref(false)
+const isAudioLoading = ref(true)
+const isPlaying = ref(false)
+const isInteractingWithSlider = ref(false)
+
+function formatToMMSS(secs: number) {
```
Comment (twynb) [inline src/frontend/components/ModalEditSegment.vue:61]: this should be moved to a composable

File: src/frontend/components/ProjectItem.vue [hunk 2]
```diff
@@ -45,14 +55,16 @@ const saveSettings = useSaveSetings()
 
 const ws = shallowRef<WaveSurfer>()
 const regions = shallowRef<Regions>()
+const blob = shallowRef<Blob>()
```
Comment (twynb) [inline src/frontend/components/ProjectItem.vue:58]: should have a more expressive name - audioBlob or something?

**Review by twynb — CHANGES_REQUESTED** [review]
LGTM, other than some minor changes.

other than those, is this still a draft or was that a misclick?

also: do i understand it correctly that the updatePeaks()/updateFilePeaks() functions and related data are used for generating the wavesurfer waveform display?

File: src/frontend/components/ModalEditSegment.vue (hunk 1 above)
Comment (chubetho) [reply src/frontend/components/ModalEditSegment.vue]: done

**Comment by chubetho** [comment]
yeah, peaks here are for waveform. They are computed in client side so they are quite expensive. So I cached them. It will be nice if peaks can be computed at time when project is created

**Comment by chubetho** [comment]
close #124

**Review by twynb — APPROVED** [review]
LGTM

**Pull request merged**